.github/workflows/build-envoy-image-ci.yaml

```diff
@@ -17,7 +17,7 @@ jobs:
         with:
           path: /tmp/buildx-cache
           key: docker-cache-${{ github.head_ref }}
-          restore-keys: docker-cache-master
+          restore-keys: docker-cache-v1.23
```

Comment:
Not sure how significant this is, but as docker caching is just a perf optimization, maybe we could just keep one cache, for whatever build was the latest? Typically that would be for a PR on top of master, and it should not hurt too much if docker cache is not effective when building for a release branch?

Reply:
Maybe we should get `ENVOY_MINOR_RELEASE` before this step to use it here?